pep/app/services/auth.ts

```diff
@@ -13,20 +34,44 @@ export interface LoginForm {
 
 export default class AuthService extends Service {
     @service('pep-session') session!: PepSessionService;
+    @service ajax!: AjaxService;
     @service modal!: ModalService;
+    @service notifications!: NotificationService;
+    @service loadingBar!: LoadingBarService;
 
     dontRedirectOnLogin = false;
 
     /**
-     * Opens the login modal dialog
+     * Opens the login modal dialog and loads the forgot password URL (along with federated login information)
      * @param {Boolean} dontRedirectOnLogin
      * @param {Object} modalOptions
      */
-    openLoginModal(dontRedirectOnLogin = false, modalOptions = {}) {
-        const model: LoginForm = { username: null, password: null };
-        const changeset = createChangeset<LoginForm>(model, LoginValidations);
-        this.dontRedirectOnLogin = dontRedirectOnLogin;
-        this.modal.open('user/login', { ...modalOptions, changeset });
+    async openLoginModal(dontRedirectOnLogin = false, modalOptions = {}) {
+        try {
+            this.loadingBar.show();
+            const model: LoginForm = { username: null, password: null };
+            const changeset = createChangeset<LoginForm>(model, LoginValidations);
+            this.dontRedirectOnLogin = dontRedirectOnLogin;
+            const session = this.session.getUnauthenticatedSession();
+            const federatedLogins = await this.ajax.request<FederatedLoginResponse>(
+                `${ENV.federatedLoginUrl}?sessionId=${session?.SessionId ?? ''}`,
```

Comment:
i would use `serializeQueryParams()` here, just for consistency in how we form urls